### ISSUE TITLE
Directory Errors on Broken Symlinks

### ISSUE THREAD

**Issue by Shinmera** [drmeister/clasp#161]
On 712260b4038b40e04d8928b7183e2af5da416a0d  
`~` contains a symlink `foo` that points to `/blablabla`, an inexistent directory.
```
> (directory "~/*")

Condition of type: SIMPLE-FILE-ERROR
In file_truename ../../src/core/unixfsys.cc line 443
COMMON-LISP:FILE-ERROR :initializers (:PATHNAME "/blablabla"  )


Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[116] TOP-LEVEL.
 File: #P"/home/linus/.cache/shirakumo/autobuild/clasp-testing/.autobuild/712260b4038b40e04d8928b7183e2af5da416a0d/build/clasp/Contents/Resources/lisp/kernel/lsp/top.lsp" (Position #21029)
```

**Comment by drmeister** [comment]
Fixed broken symlink behavior in 093e9bf of directory, probe-file and truename.  Broken symlinks are treated like regular files the way SBCL does.

**Comment by Shinmera** [comment]
Fix confirmed on 9c24f9a45d0bd4bd4a80cdcac18b7cee012947fb
```
> (directory "~/*")

(#P"/home/linus/.zsh-update" #P"/home/linus/.mkclrc" #P"/home/linus/.bashrc"
 #P"/home/linus/.gitconfig" #P"/home/linus/.bash_history" #P"/home/linus/foo"
 #P"/home/linus/.xinitrc" #P"/home/linus/.sbclrc" #P"/home/linus/.bash_logout"
 #P"/home/linus/.xfigrc" #P"/home/linus/.zshrc" #P"/home/linus/.bash_profile"
 #P"/home/linus/.Xdefaults" #P"/home/linus/.Xauthority" #P"/home/linus/.eclrc"
 #P"/home/linus/.zcompdump" #P"/home/linus/.history" #P"/home/linus/.abclrc"
 #P"/home/linus/.emacs")
```
